pkg/controller/kieapp/defaults/defaults_test.go

```diff
@@ -6035,3 +6035,308 @@ func createKafkaJbpmObject(dateFormat string, tasksTopics string, casesTopics st
+					},
+				},
+			},
+		},
+	}
+	env, err := GetEnvironment(cr, test.MockService())
+	assert.Nil(t, err, "Error getting test-cors Test environment")
+	assert.NotNil(t, env)
+	checkKieServerCORSAssertions(t, cr, env)
+	cors := cr.Status.Applied.Objects.Servers[0].Cors
+	assert.NotNil(t, cors)
+	checkCors(t, cors)
+	checkConsoleCORSAssertions(t, cr, env)
+}
+
+func TestEnvironmentWithPartialCORS(t *testing.T) {
+	const name = "test-cors"
+	cr := &api.KieApp{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: api.KieAppSpec{
+			Environment: api.RhpamTrial,
+			Objects: api.KieAppObjects{
+				Console: &api.ConsoleObject{
+					Cors: getPartialCors(),
+				},
+				Servers: []api.KieServerSet{
+					{
+						Cors: getPartialCors(),
+					},
+				},
+			},
+		},
+	}
+	env, err := GetEnvironment(cr, test.MockService())
+	assert.Nil(t, err, "Error getting test-cors Test environment")
+	assert.NotNil(t, env)
+	checkKieServerCORSAssertions(t, cr, env)
+	cors := cr.Status.Applied.Objects.Servers[0].Cors
+	assert.NotNil(t, cors)
+	checkCustomCors(t, cors)
+	checkConsoleCustomCORSAssertions(t, cr, env)
+}
+
+func TestTrialEnvironmentWithCORSDefault(t *testing.T) {
+	const name = "test-cors-default"
+	cr := &api.KieApp{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: api.KieAppSpec{
+			Environment: api.RhpamTrial,
+			Objects: api.KieAppObjects{
+				Console: &api.ConsoleObject{
+					Cors: &api.CORSFiltersObject{},
+				},
+				Servers: []api.KieServerSet{
+					{
+						Cors: &api.CORSFiltersObject{},
+					},
+				},
+			},
+		},
+	}
+	env, err := GetEnvironment(cr, test.MockService())
+	assert.Nil(t, err, "Error getting test-cors-default Test environment")
+	assert.NotNil(t, env)
+	cors := cr.Status.Applied.Objects.Console.Cors
+	assert.Equal(t, cors.Filters, "")
+	cors = cr.Status.Applied.Objects.Servers[0].Cors
+	assert.Equal(t, cors.Filters, "")
+}
+
+func TestDashbuilderWithoutCORS(t *testing.T) {
+	const name = "test-no-cors-dashbuilder"
+	cr := &api.KieApp{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: api.KieAppSpec{
+			Environment: api.RhpamStandaloneDashbuilder,
+			Objects: api.KieAppObjects{
+				Dashbuilder: &api.DashbuilderObject{},
+			},
+		},
+	}
+	env, err := GetEnvironment(cr, test.MockService())
+	assert.Nil(t, err, "Error getting test-no-cors-dashbuilder Test environment")
+	assert.NotNil(t, env)
+	corsEnabled := isCORSEnabled(env.Dashbuilder.DeploymentConfigs[0].Spec.Template.Spec.Containers[0].Env)
+	assert.False(t, corsEnabled)
+	assert.Empty(t, cr.Spec.Objects.Dashbuilder)
+	assert.Nil(t, cr.Status.Applied.Objects.Dashbuilder.Cors)
+}
+
+func TestDashbuilderWithCORS(t *testing.T) {
+	const name = "test-cors-dashbuilder"
+	cr := &api.KieApp{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: api.KieAppSpec{
+			Environment: api.RhpamStandaloneDashbuilder,
+			Objects: api.KieAppObjects{
+				Dashbuilder: &api.DashbuilderObject{
+					Cors: getDefaultCors(),
```

Comment:
how user will be adding the default on the CR?

Reply:
Adding a boolean flag to enable default values